### PR TITLE
Revocation E2E test

### PR DESCRIPTION
**Pull request by shitz**
This PR adds an integration test for revocation. To that end, the test clients have been extended to handle revocations and optionally retry the test packet up to N times with different paths. The test first runs one round of the e2e integration test between all pairs (without any retries of packets), then introduces link failures between 2-23 <-> 2-26 and 1-11 <-> 2-21 (by stopping br2-26-2 and br1-11-3) and runs another round of the e2e integration test between all pairs, retrying a test packet up to 3 times. The failed links are chosen in a way that ensures at least one working path between any two ASes. The test is thus successful, if every AS can still communicate with every other AS (like with the e2e integration test). 

Note: This test is specifically tailored to our default test topo and probably has to be changed if the topo ever changes.

It's not a perfect test, but it's better than having nothing at all for a core functionality of SCION.

depends on #947

---
This change is [Reviewable](https://reviewable.io/reviews/netsec-ethz/scion/950)

### CODE REVIEW

**Comment by klausman** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[test/integration/base_cli_srv.py, line 188 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbN5ufcRWJEpcdHN1:-K_cbN5ufcRWJEpcdHN2:b8kbzk3) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L188)):*
> ```Python
>             if not spkt:
>                 if self.retries:
>                     logging.info("Timeout waiting for response. Retrying...")
> ```

This should log how many retries remain.

---

*[test/integration/revocation_test.sh, line 27 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbv_4is2kbe9sAROY:-K_cbv_5y9ZyN294kyuL:b-8rak4k) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L27)):*
> ```Shell
> result=$?
> if [ ${result} -ne 0 ]; then
>     log "E2E test failed."
> ```

Log the result value here.

---

*[test/integration/revocation_test.sh, line 41 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cc0oSbTxlcGGF1CUD:-K_cc0oSbTxlcGGF1CUE:b-yz0ec1) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L41)):*
> ```Shell
> result=$?
> if [ $result -ne 0 ]; then
>     log "E2E test with failed routers failed."
> ```

Same as comment above (retval logging).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by kormat** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[docker/integration_test.sh, line 67 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_c_JQ8BV6HM8Ew5wwz:-K_c_JQ92hBp9qdtwc87:bmyb6i6) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/docker/integration_test.sh#L67)):*
> ```Shell
> 
> run Revocation test/integration/revocation_test.sh
> result=$((${result}+$?))
> ```

This can be simplified to `result=$((result+$?))`

---

*[endhost/sciond.py, line 240 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_caEgw85VC67C-oLkJ:-K_caEgw85VC67C-oLkK:be1knrb) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L240)):*
> ```Python
> 
>     def handle_revocation(self, rev_info, meta):
>         assert isinstance(rev_info, RevocationInfo)
> ```

Does this work for both control messages and SCMP messages?

---

*[endhost/sciond.py, line 266 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_caoti8P5ezvhJ6u1C:-K_caoti8P5ezvhJ6u1D:bpm7jsg) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L266)):*
> ```Python
>                           "current epoch %d." %
>                           (rev_info.p.epoch,
>                            ConnectedHashTree.get_current_epoch()))
> ```

This formatting becomes a Lot better if you move the first arg down to a new line.

---

*[endhost/sciond.py, line 272 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cb3o2V9K6svwxLLrZ:-K_cb3o2V9K6svwxLLr_:b-7nexaw) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L272)):*
> ```Python
>         for segment in db(full=True):
>             for asm in segment.iter_asms():
>                 if self.verify_asm(asm, rev_info):
> ```

Oh, just realised this name is terrible. Can you rename it as well?

---

*[test/integration/base_cli_srv.py, line 193 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_ccWa07vh_mZGIBWUW:-K_ccWa07vh_mZGIBWUX:bgim2y7) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L193)):*
> ```Python
>                 else:
>                     logging.error("Timeout waiting for response.")
>                     self._stop(success=False)
> ```

This is the default value of `success`).

---

*[test/integration/base_cli_srv.py, line 199 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbtK2EUtHBhCHNt9s:-K_cbtK2EUtHBhCHNt9t:b-jt5dz7) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L199)):*
> ```Python
>                     self._stop(success=bool(r_code))
>                 else:
>                     sleep_dur = max(0, 1.0 - recv_dur)
> ```

Is this logic different than `util.sleep_interval`?

---

*[test/integration/end2end_test.py, line 77 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_ceAqfQABWe24FPxzL:-K_ceAqg4cYtPL5mUdFQ:b-w7l33v) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/end2end_test.py#L77)):*
> ```Python
>             rev_info = RevocationInfo.from_raw(scmp_pld.info.rev_info)
>             logging.info("Received revocation for IF %d." % rev_info.p.ifID)
> ```

blank line

---

*[test/integration/end2end_test.py, line 79 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cdnOnRS2pD3U-QjMy:-K_cdnOnRS2pD3U-QjMz:b-j19lhg) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/end2end_test.py#L79)):*
> ```Python
> 
>             self.sd.handle_revocation(rev_info, None)
>             if not self.retries:
> ```

The `retries` handling logic is spread out, and it's difficult to determine if it work correctly in all cases. It should all be in `TestClientBase.run()`.

---

*[test/integration/revocation_test.sh, line 32 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cfM634kQa6uOUlyF9:-K_cfM634kQa6uOUlyFA:b-c74ksr) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L32)):*
> ```Shell
> # Bring down routers.
> SLEEP=10
> log "Stopping routers and waiting for ${SLEEP}s."
> ```

This will only work with the default topo, which is a problem. Until now, `docker/integration_test.sh` would work with any topology. E.g. this change makes it break for Tiny.topo. Maybe add a check to make sure the specified BR's exist, and exit 0 with a log message saying skipping the test if they don't. Also consider adding cmdline args for the BRs, so that it can be run manually against different topologies.

---

*[test/integration/revocation_test.sh, line 33 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cekBFfNDeGcrGIk-T:-K_cekBFfNDeGcrGIk-U:bqlakb0) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L33)):*
> ```Shell
> SLEEP=10
> log "Stopping routers and waiting for ${SLEEP}s."
> supervisorctl -s http://localhost:9011 stop as1-11:br1-11-3 > /dev/null
> ```

Use `./supervisor/supervisor.sh`, that's what it's for.  You can also stop both in one line.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by shitz** [comment]
Review status: 0 of 7 files reviewed at latest revision, 13 unresolved discussions.

---

*[docker/integration_test.sh, line 67 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_c_JQ8BV6HM8Ew5wwz:-K_ck2hbt-YQCbIlDL9G:b-ir54fc) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/docker/integration_test.sh#L67)):*
> esult=$((result+$?))
I thought it was best practice to always use `${}` around vars. Changed it.

---

*[endhost/sciond.py, line 240 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_caEgw85VC67C-oLkJ:-K_clOCPZKoFUBLBdVvx:bt6caeh) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L240)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Does this work for both control messages and SCMP messages?
</blockquote></details>

Good point. As it stands it wouldn't work for SCMP revocations received by the SCIOND directly (which wasn't a problem, since the client calls SCIOND's `handle_revocation` directly due to the lack of a protocol between application and SCIOND. Fixed.

---

*[endhost/sciond.py, line 266 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_caoti8P5ezvhJ6u1C:-K_cmlv2dawkRfA7yVBX:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L266)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This formatting becomes a Lot better if you move the first arg down to a new line.
</blockquote></details>

Done.

---

*[endhost/sciond.py, line 272 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cb3o2V9K6svwxLLrZ:-K_cmnYUhulsC5h6-Q9V:bmo13vk) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/endhost/sciond.py#L272)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Oh, just realised this name is terrible. Can you rename it as well?
</blockquote></details>

I agree and I renamed it in the follow up PR. Moved renaming to this PR.

---

*[test/integration/base_cli_srv.py, line 188 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbN5ufcRWJEpcdHN1:-K_cnkEnxphLALT4dmwd:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L188)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

This should log how many retries remain.
</blockquote></details>

Done.

---

*[test/integration/base_cli_srv.py, line 193 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_ccWa07vh_mZGIBWUW:-K_cnr69w2eYSfI6KAoT:bv1opgx) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L193)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This is the default value of `success`).
</blockquote></details>

I know, but I thought it would add some readability. In that case the default arg wouldn't be needed at all. Changed to your suggestion.

---

*[test/integration/base_cli_srv.py, line 199 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbtK2EUtHBhCHNt9s:-K_cp8CLwK3aqJwTxbID:b-jlrrbq) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/base_cli_srv.py#L199)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Is this logic different than `util.sleep_interval`?
</blockquote></details>

Code changed, s.t. `util.sleep_interval` isn't really fitting anymore.

---

*[test/integration/end2end_test.py, line 77 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_ceAqfQABWe24FPxzL:-K_cpHLzwal3FYx78yBG:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/end2end_test.py#L77)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

blank line
</blockquote></details>

Done.

---

*[test/integration/end2end_test.py, line 79 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cdnOnRS2pD3U-QjMy:-K_cvCoUe1TLJR9mP01E:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/end2end_test.py#L79)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

The `retries` handling logic is spread out, and it's difficult to determine if it work correctly in all cases. It should all be in `TestClientBase.run()`.
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 27 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cbv_4is2kbe9sAROY:-K_cvjbUSaV3-G8kQnHa:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L27)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Log the result value here.
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 32 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cfM634kQa6uOUlyF9:-K_d7lN7SgVdiwORgnu_:b-23mtrl) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L32)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This will only work with the default topo, which is a problem. Until now, `docker/integration_test.sh` would work with any topology. E.g. this change makes it break for Tiny.topo. Maybe add a check to make sure the specified BR's exist, and exit 0 with a log message saying skipping the test if they don't. Also consider adding cmdline args for the BRs, so that it can be run manually against different topologies.
</blockquote></details>

Added functionality to check if the BRs exist and pass BRs as command line arguments.

---

*[test/integration/revocation_test.sh, line 33 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cekBFfNDeGcrGIk-T:-K_cw1SilCVhf7m3o2qN:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L33)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Use `./supervisor/supervisor.sh`, that's what it's for.  You can also stop both in one line.
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 41 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cc0oSbTxlcGGF1CUD:-K_cvoqznEslToQ-zFb4:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L41)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Same as comment above (retval logging).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by klausman** [comment]
Review status: 0 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[docker/integration_test.sh, line 67 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_c_JQ8BV6HM8Ew5wwz:-K_dFSRsF5vZP0Y_pxY6:b6mbfqr) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/docker/integration_test.sh#L67)):*
<details><summary><i>Previously, shitz wrote…</i></summary><blockquote>

> esult=$((result+$?))
I thought it was best practice to always use `${}` around vars. Changed it.
</blockquote></details>

Re: `${}`: In most contexts, `${}` prevents accidental inclusion of subsequent characters into the var name. Since it's not always intuitive what bash may consider part of the name, using it almost always is a reasonable piece of advice.

However, `${}` does not come for free: it's more work to parse for bash and --- in extremely rare corner cases --- may backfire.

For arithmetic expressions (i.e. `$(())`), the set of allowed non-varname characters is already limited, causing an obvious error message when what `${}` tries to prevent happens.

tl;dr: in `$(())` the `${}` idiom is usually not needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by kormat** [comment]
Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[docker/integration_test.sh, line 67 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_c_JQ8BV6HM8Ew5wwz:-K_dEul7V-HHcnPQD0JE:b-ef0tzj) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/docker/integration_test.sh#L67)):*
<details><summary><i>Previously, shitz wrote…</i></summary><blockquote>

> esult=$((result+$?))
I thought it was best practice to always use `${}` around vars. Changed it.
</blockquote></details>

Eh, it's good to use them when it's ambiguous, but the `$((...))` construct is actually looking for var names, so there's no need to add explicit referencing.

---

*[docker/integration_test.sh, line 66 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dHVNx80maDEKl-zOn:-K_dHVNydtKf2kKKYmt4:bsifgdj) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/docker/integration_test.sh#L66)):*
> ```Shell
> result=$?
> 
> run Revocation test/integration/revocation_test.sh as1-11:br1-11-3 as2-26:br2-26-2
> ```

Suggestion:
```
run Revocation test/integration/revocation_test.sh ${REV_BRS:-as1-11:br1-11-3 as2-26:br2-26-2}
```

This means that to run against different BRs, someone can do `REV_BRS="as1-19:br1-19-5" ./docker/integration_test.sh`, for example.

---

*[test/integration/base_cli_srv.py, line 189 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dISG3u1zGHlHtruYR:-K_dISG44l7oW128Qa0h:bw75vk0) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/base_cli_srv.py#L189)):*
> ```Python
>                 logging.info("Timeout waiting for response")
>                 self._retry_or_stop()
>             else:
> ```

If the previous `if` block ends with `continue`, this `else` isn't needed.

---

*[test/integration/base_cli_srv.py, line 194 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dIcw9aVXR1EeMwy70:-K_dIcw9aVXR1EeMwy71:b-apq1jj) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/base_cli_srv.py#L194)):*
> ```Python
>                     self._stop(success=bool(r_code))
>                 else:
>                     self._retry_or_stop(1.0 - recv_dur)
> ```

A comment explaining the `1.0` would be good.

---

*[test/integration/revocation_test.sh, line 32 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cfM634kQa6uOUlyF9:-K_dGjkO9w_3eIr5T7e8:bouyibi) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L32)):*
<details><summary><i>Previously, shitz wrote…</i></summary><blockquote>

Added functionality to check if the BRs exist and pass BRs as command line arguments.
</blockquote></details>

Ok, but now the defaults are hardcoded in `docker/integration_test.sh` instead. I'll follow-up there.

---

*[test/integration/revocation_test.sh, line 33 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cekBFfNDeGcrGIk-T:-K_dH0yX6ae5GMLMYmsb:bv9d8pj) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L33)):*
<details><summary><i>Previously, shitz wrote…</i></summary><blockquote>

Done.
</blockquote></details>

You need to check the return value, otherwise errors are completely silent (e.g. a BR has already crashed). I would probably also not redirect output.

---

*[test/integration/revocation_test.sh, line 30 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGS4vTUc5dvMiwT8W:-K_dGS4wS44CzrQU13b3:bfe1fh0) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L30)):*
> ```Shell
> }
> 
> for br in $@; do
> ```

This needs to be `"$@"`

---

*[test/integration/revocation_test.sh, line 31 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGVprG6oJEGyXpMly:-K_dGVprG6oJEGyXpMlz:b7ztdhf) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L31)):*
> ```Shell
> 
> for br in $@; do
>     check_br_exists $@
> ```

This should be `"$br"`

---

*[test/integration/revocation_test.sh, line 32 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGZKjryTMj1Oy8Z3p:-K_dGZKjryTMj1Oy8Z3q:bwj3lxx) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L32)):*
> ```Shell
> for br in $@; do
>     check_br_exists $@
>     if [ $? -eq 1 ]; then
> ```

No reason you can't do the check in one go. `if ! check_br_exists "$br"; then`

---

*[test/integration/revocation_test.sh, line 49 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dHIVdYnP-n-7nvxxv:-K_dHIVdYnP-n-7nvxxw:b7z621x) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L49)):*
> ```Shell
> SLEEP=10
> log "Stopping routers and waiting for ${SLEEP}s."
> ./supervisor/supervisor.sh stop $@ > /dev/null
> ```

This should be `"$@"`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by shitz** [comment]
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[docker/integration_test.sh, line 67 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_c_JQ8BV6HM8Ew5wwz:-K_dSifABa1nUdp7mGRm:bdzp4b4) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/docker/integration_test.sh#L67)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Eh, it's good to use them when it's ambiguous, but the `$((...))` construct is actually looking for var names, so there's no need to add explicit referencing.
</blockquote></details>

Thanks for the info. As you can see, my level of bash expertise is basically copy & pasting from SO/existing code.

---

*[docker/integration_test.sh, line 66 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dHVNx80maDEKl-zOn:-K_dTHHaYUuAuPSRIYYr:bkoke9u) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/docker/integration_test.sh#L66)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Suggestion:
```
run Revocation test/integration/revocation_test.sh ${REV_BRS:-as1-11:br1-11-3 as2-26:br2-26-2}
```

This means that to run against different BRs, someone can do `REV_BRS="as1-19:br1-19-5" ./docker/integration_test.sh`, for example.
</blockquote></details>

I like it. Done.

---

*[test/integration/base_cli_srv.py, line 189 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dISG3u1zGHlHtruYR:-K_dU-aY7tHuk1lslZyY:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/base_cli_srv.py#L189)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

If the previous `if` block ends with `continue`, this `else` isn't needed.
</blockquote></details>

Done.

---

*[test/integration/base_cli_srv.py, line 194 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dIcw9aVXR1EeMwy70:-K_dU0DWsoPXqSbwU63s:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/base_cli_srv.py#L194)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

A comment explaining the `1.0` would be good.
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 33 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_cekBFfNDeGcrGIk-T:-K_dSecW_8eeIqjFJFzZ:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/8dc8502f5cd5450f493cda1b8d545547349ca995/test/integration/revocation_test.sh#L33)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

You need to check the return value, otherwise errors are completely silent (e.g. a BR has already crashed). I would probably also not redirect output.
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 30 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGS4vTUc5dvMiwT8W:-K_dSD9lLbhFTSveiyMj:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L30)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This needs to be `"$@"`
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 31 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGVprG6oJEGyXpMly:-K_dSDUgydFxiSPsmkex:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L31)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This should be `"$br"`
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 32 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dGZKjryTMj1Oy8Z3p:-K_dSK8ecdRulbWGWJL7:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L32)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

No reason you can't do the check in one go. `if ! check_br_exists "$br"; then`
</blockquote></details>

Done.

---

*[test/integration/revocation_test.sh, line 49 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dHIVdYnP-n-7nvxxv:-K_dShDjWRhdUanhHIiv:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/935f19519ae6ac9e518127f87b4441f43919575b/test/integration/revocation_test.sh#L49)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

This should be `"$@"`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950)*

**Comment by kormat** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[docker/integration_test.sh, line 66 at r3](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dw4pGBSlKUtfrzmb5:-K_dw4pGBSlKUtfrzmb6:b-53vzi3) ([raw file](https://github.com/netsec-ethz/scion/blob/00843e59c356faba7a4d2bac4066e63da010bf7a/docker/integration_test.sh#L66)):*
> ```Shell
> result=$?
> 
> run Revocation "test/integration/revocation_test.sh\
> ```

Don't bother line-wrapping this. Shell tends to have long lines, and only sometimes does wrapping them make it easier to read.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-:-K_dyQwVd8TPIct6QoIq:bnfp4nl)*

**Comment by shitz** [comment]
Thanks for the review!

---

Review status: 6 of 7 files reviewed at latest revision, 5 unresolved discussions.

---

*[docker/integration_test.sh, line 66 at r3](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-K_dw4pGBSlKUtfrzmb5:-K_e3RpVXUP2c0NmdXPD:b-896fix) ([raw file](https://github.com/netsec-ethz/scion/blob/00843e59c356faba7a4d2bac4066e63da010bf7a/docker/integration_test.sh#L66)):*
<details><summary><i>Previously, kormat (Stephen Shirley) wrote…</i></summary><blockquote>

Don't bother line-wrapping this. Shell tends to have long lines, and only sometimes does wrapping them make it easier to read.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion/950#-:-K_e3T7hJlxRsGY1dh8Z:b-9qdoz3)*
